hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -278,14 +289,19 @@ protected Option<DeleteRecord> doProcessNextDeletedRecord(DeleteRecord deleteRec
           if (isDeleteRecordWithNaturalOrder(existingRecordMetadataPair.getLeft(), existingOrderingVal)) {
             return Option.empty();
           }
-          Comparable deleteOrderingVal = deleteRecord.getOrderingValue();
-          // Checks the ordering value does not equal to 0
-          // because we use 0 as the default value which means natural order
-          boolean chooseExisting = !deleteOrderingVal.equals(0)
-              && ReflectionUtils.isSameClass(existingOrderingVal, deleteOrderingVal)
-              && existingOrderingVal.compareTo(deleteOrderingVal) > 0;
+          Comparable deleteOrderingVal = readerContext.getOrderingValue(
```

Comment:
can we define a boolean value named "orderingValueMissing" and so code readabiltiy is better.

Reply:
Yeah, we can do that.

Reply:
I use default value directly, and create a function to indicate if we found a processing time delete.